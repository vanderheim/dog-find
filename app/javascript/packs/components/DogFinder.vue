<template>
<v-card class="mx-auto my-12" max-width="600">
    <v-img aspect-ratio="1" height="600" :src='dog.message' />


    <v-card-text class="text-center">

        <v-snackbar v-model="createdNotification" color='success'>
            You just added a new dog to your collection!

                <v-btn outlined text @click="createdNotification = false">
                    CLOSE
                </v-btn>
        </v-snackbar>

        <v-btn x-large color="success" @click="save(dog.message)">
            LOVE IT!
        </v-btn>
        <v-btn x-large color="error" @click="fetchDog">
            NO THANKS
        </v-btn>

    </v-card-text>
</v-card>
</template>

<script>
import axios from "axios";
export default {
    data: () => ({
        createdNotification: false,
        dog: {}
    }),

    created() {
        this.initialize();
    },

    methods: {

        fetchDog() {
            return axios
                .get("https://dog.ceo/api/breed/maltese/images/random")
                .then(response => {
                    console.log(response.data);
                    this.dog = response.data;
                })
                .catch(e => {
                    console.log(e);
                });
        },

        initialize() {
            return axios
                .get("https://dog.ceo/api/breed/maltese/images/random")
                .then(response => {
                    console.log(response.data);
                    this.dog = response.data;
                })
                .catch(e => {
                    console.log(e);
                });
        },

        save(item) {
            this.createdNotification = true;
            console.log(item);

            const dog = {
                url: item
            };

            axios
                .post(`/dogs/`, {
                    dog: dog
                })
                .then(response => {
                    this.initialize();
                })
                .catch(error => {
                    console.log(error);
                });
        },

        close() {
            this.dialog = false;
        }
    }
};
</script>
