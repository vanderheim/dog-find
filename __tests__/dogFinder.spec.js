import { createLocalVue, mount } from '@vue/test-utils'
import DogFinder from '../app/javascript/packs/components/DogFinder.vue'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import flushPromises from "flush-promises";

// Mock axios call to receive the image of the dog.
jest.mock("axios", () => ({
    get: () => Promise.resolve({
        data: [
            { message: 'success.jpg' }
        ]
    })
}));

describe('DogFinder', () => {
    const localVue = createLocalVue();

    // Initialize vue router on local instance.
    localVue.use(VueRouter);
    const router = new VueRouter();

    // Initialize vuetify on local instance.
    let vuetify;
    beforeEach(() => {
        vuetify = new Vuetify()
    });

    // Mount function to use repeatedly for mounting Vue instance.
    const mountFunction = options => {
        return mount(DogFinder, {
            localVue,
            vuetify,
            router,
            ...options,
        })
    };

    it('received dog image from mock api call', async () => {
        const wrapper = mountFunction();
        await flushPromises();
        expect(wrapper.vm.dog[0].message).toBe('success.jpg');
    });

    it('should match snapshot', () => {
        const wrapper = mountFunction();
    
        expect(wrapper.html()).toMatchSnapshot()
      });
})