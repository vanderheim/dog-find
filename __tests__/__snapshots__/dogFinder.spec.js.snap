// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogFinder should match snapshot 1`] = `
<div class="mx-auto my-12 v-card v-sheet theme--light" style="max-width: 600px;">
  <div class="v-image v-responsive theme--light" style="height: 600px;">
    <div class="v-responsive__sizer" style="padding-bottom: 100%;"></div>
    <div class="v-responsive__content"></div>
  </div>
  <div class="v-card__text text-center">
    <div class="v-snack v-snack--bottom v-snack--has-background" style="padding-bottom: 0px; padding-top: 0px;">
      <transition-stub name="v-snack-transition">
        <div class="v-snack__wrapper v-sheet theme--dark success" style="display: none;">
          <div role="status" aria-live="polite" class="v-snack__content">
            You just added a new dog to your collection!

            <button type="button" class="v-btn v-btn--outlined v-btn--text theme--dark v-size--default"><span class="v-btn__content">
                    CLOSE
                </span></button>
          </div>
          <div class="v-snack__action "></div>
        </div>
      </transition-stub>
    </div> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--x-large success"><span class="v-btn__content">
            LOVE IT!
        </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--x-large error"><span class="v-btn__content">
            NO THANKS
        </span></button>
  </div>
</div>
`;
